tests/garage/torch/policies/test_gaussian_mlp_policy.py

```diff
@@ -0,0 +1,100 @@
+import pickle
+from unittest import mock
+
+import numpy as np
+import pytest
+import torch
+
+from garage.torch.policies import GaussianMLPPolicy
+
+
+@mock.patch('garage.torch.policies.gaussian_mlp_policy.GaussianMLPModule')
+def test_policy_get_actions(mock_model):
+    action = torch.Tensor([0.5, 0.1, 0.6])
+    mean = torch.Tensor([1., 2., 3.])
+    variance = torch.Tensor([0.1, 0.2, 0.3])
+
+    mock_dist = mock.MagicMock()
+    mock_dist.rsample.return_value = action
+    mock_dist.mean = mean
+    mock_dist.variance = variance
+
+    mock_model.return_value = lambda x: mock_dist
+    input_dim, output_dim, hidden_sizes = (5, 3, (2, 2))
+
+    env_spec = mock.MagicMock()
+    env_spec.observation_space.flat_dim = input_dim
+    env_spec.action_space.flat_dim = output_dim
+
+    policy = GaussianMLPPolicy(env_spec, hidden_sizes=hidden_sizes)
+
+    input = torch.ones(input_dim)
+    sample, dist_info = policy.get_actions(input)
+
+    assert np.array_equal(sample, action.detach().numpy())
+    assert np.array_equal(dist_info['mean'], mean.detach().numpy())
+    assert np.array_equal(dist_info['log_std'], variance.sqrt().detach().numpy())
+
+
+@mock.patch('garage.torch.policies.gaussian_mlp_policy.GaussianMLPModule')
+def test_policy_get_action(mock_model):
+    action = torch.Tensor([0.5, 0.1, 0.6])
+    mean = torch.Tensor([1., 2., 3.])
+    variance = torch.Tensor([0.1, 0.2, 0.3])
+
+    mock_dist = mock.MagicMock()
+    mock_dist.rsample.return_value = action
+    mock_dist.mean = mean
+    mock_dist.variance = variance
+
+    mock_model.return_value = lambda x: mock_dist
+    input_dim, output_dim, hidden_sizes = (5, 3, (2, 2))
+
+    env_spec = mock.MagicMock()
+    env_spec.observation_space.flat_dim = input_dim
+    env_spec.action_space.flat_dim = output_dim
+
+    policy = GaussianMLPPolicy(env_spec, hidden_sizes=hidden_sizes)
+
+    input = torch.ones(input_dim)
+    sample, dist_info = policy.get_action(input)
+
+    assert sample == action[0].item()
+    assert dist_info['mean'] == mean[0].item()
+    assert dist_info['log_std'] == variance[0].sqrt().item()
+
+
+@pytest.mark.parametrize(
+    "input_dim,output_dim,hidden_sizes",
+    [(5, 1, (1,)),
+     (5, 1, (2,)),
+     (5, 2, (3,)),
+     (5, 2, (1, 1)),
+     (5, 3, (2, 2))])
+def test_policy_is_picklable(input_dim, output_dim, hidden_sizes):
+    env = TestEnv(TestSpace(input_dim), TestSpace(output_dim))
+
+    policy = GaussianMLPPolicy(env, hidden_sizes=hidden_sizes)
+
+    input = torch.ones(input_dim)
+    sample, dist_info = policy.get_actions(input)
+
+    h = pickle.dumps(policy)
+    policy_pickled = pickle.loads(h)
+
+    sample_p, dist_info_p = policy_pickled.get_actions(input)
+
+    assert np.array_equal(sample.shape, sample_p.shape)
+    assert np.array_equal(dist_info['mean'], dist_info_p['mean'])
+    assert np.array_equal(dist_info['log_std'], dist_info_p['log_std'])
+
+
+class TestEnv():
```

Comment:
there are existing fixtures like this in tests/fixtures. you can also use Mock (preferable)

Reply:
I tried to use the Mock, but if I used mock object as a test_env, 
I got the error 
_pickle.PicklingError: Can't pickle <class 'unittest.mock.MagicMock'>: it's not the same object as unittest.mock.MagicMock

Reply:
delete this code because the pickable testcase is removed